packages/eslint-plugin/src/rules/comma-spacing.ts

```diff
@@ -0,0 +1,190 @@
+import {
+  TSESTree,
+  AST_TOKEN_TYPES,
+} from '@typescript-eslint/experimental-utils';
+import { isClosingParenToken, isCommaToken } from 'eslint-utils';
```

Comment:
we should update `eslint-utils.d.ts` to return type guards instead of booleans,

eg: 
```
export function isCommaToken(
  token: TSESTree.Token | TSESTree.Comment,
): boolean;
```
to something like this
```
export function isCommaToken(
   token: TSESTree.Token | TSESTree.Comment,
): token is TSESTree.Token;
```

Reply:
edit, this can be done in separate PR to not pollute this one

Reply:
@armano2 Oke, will do it in new PR.

Reply:
change was not that simple as Token and Comment was a little mixed up
#1497